src/lib.rs

```diff
@@ -168,7 +168,7 @@ pub fn firefox_version(binary: &Path) -> Result<AppVersion, Error>  {
     let mut version = AppVersion::new();
     let mut updated = false;
 
-    if let Some(dir) = ini_path(binary) {
+    if let Some(dir) = ini_path(&binary.canonicalize().unwrap()) {
```

Comment:
A [`Result`](https://doc.rust-lang.org/std/result/enum.Result.html) can either be `Ok(T)` or `Err(E)`, and [`res.unwrap()`](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap) tries to get `T` but if `canonicalize` was not OK, it panics the program.